src/Volume.jsx

```diff
@@ -2,27 +2,67 @@
 import { TrashIcon } from '@patternfly/react-icons';
 import { FileAutoComplete } from 'cockpit-components-file-autocomplete.jsx';
 import cockpit from 'cockpit';
 
+import * as utils from './util.js';
+
 const _ = cockpit.gettext;
 
-export const Volume = ({ id, item, onChange, idx, removeitem, additem, options, itemCount }) =>
+export function validateVolume(value, key, validationFailed) {
+    const delta = { ...validationFailed };
+    delete delta[key];
+
+    switch (key) {
+    case "hostPath":
+        break;
+    case "containerPath":
+        if (utils.isEmpty(value))
+            delta.containerPath = _("Container path must not be empty");
+
+        break;
+    }
+
+    return delta;
+}
+
+export const Volume = ({ id, item, onChange, idx, removeitem, additem, options, itemCount, validationFailed, onValidationChange }) =>
     (
         <Grid hasGutter id={id}>
-            <FormGroup className="pf-m-4-col-on-md" label={_("Host path")} fieldId={id + "-host-path"}>
+            <FormGroup className="pf-m-4-col-on-md"
+                id={id + "-host-path-group"}
+                label={_("Host path")}
+                fieldId={id + "-host-path"}
+            >
                 <FileAutoComplete id={id + "-host-path"}
                     value={item.hostPath || ''}
-                    onChange={ value => onChange(idx, 'hostPath', value) } />
+                    onChange={value => {
+                        utils.validationClear(validationFailed, "hostPath", onValidationChange);
+                        utils.validationDebounce(() => onValidationChange(validateVolume(value, "hostPath", validationFailed)));
+                        onChange(idx, 'hostPath', value);
+                    }} />
+                <FormHelper helperTextInvalid={validationFailed?.hostPath} />
             </FormGroup>
-            <FormGroup className="pf-m-3-col-on-md" label={_("Container path")} fieldId={id + "-container-path"}>
+            <FormGroup className="pf-m-3-col-on-md"
+                id={id + "-container-path-group"}
+                label={_("Container path")}
+                fieldId={id + "-container-path"}
+                isRequired
+            >
                 <TextInput id={id + "-container-path"}
                     value={item.containerPath || ''}
-                    onChange={(_event, value) => onChange(idx, 'containerPath', value)} />
+                    validated={validationFailed?.containerPath ? "error" : "default"}
+                    onChange={(_event, value) => {
+                        utils.validationClear(validationFailed, "containerPath", onValidationChange);
+                        utils.validationDebounce(() => onValidationChange(validateVolume(value, "containerPath", validationFailed)));
+                        onChange(idx, 'containerPath', value);
+                    }} />
+                <FormHelper helperTextInvalid={validationFailed?.containerPath} />
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20231016-121055-6688b0e7-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)